compliance_framework/types.py

```diff
@@ -0,0 +1,106 @@
+"""Type definitions and metadata structures for the compliance framework."""
+from dataclasses import dataclass
+from typing import Dict, List, Union, TypeVar, Callable, Any, Optional
+from datetime import datetime
+from enum import Enum
+
+# Type definitions
+RequirementId = str
+RiskId = str
+RiskIds = List[RiskId]
```

Comment:
(minor) not sure type aliasing is worth it just for collection types, I'd find it clearer to see `list[RiskId]` (as you have used below) or even just `list[str]` when viewing types in my editor